### ISSUE TITLE
[BUG] mlflow autolog failed to log xgboost metrics like train_mse, train_mae etc

### ISSUE THREAD

**Issue by PytLab** [mlflow/mlflow#7132]
### Issues Policy acknowledgement

- [X] I have read and agree to submit bug reports in accordance with the [issues policy](https://www.github.com/mlflow/mlflow/blob/master/ISSUE_POLICY.md)

### Willingness to contribute

Yes. I would be willing to contribute a fix for this bug with guidance from the MLflow community.

### MLflow version

1.26.0

### System information

- **CentOS Linux 7 (Core)**:
- **3.9.1**:

### Describe the problem

I use `mlflow.autolog` before the xgboost model training, but no `train_mse`, `train_mae` metrics tracked in the mlflow UI
<img width="1003" alt="image" src="https://user-images.githubusercontent.com/8553710/197146029-012f934e-3a76-4b5f-8341-f7827d63581a.png">

### Tracking information

MLflow version: 1.26.0
Tracking URI: file:///home/xxx/pipeline_runs/lite/debug/mlruns
Artifact URI: file:///home/xxx/pipeline_runs/lite/debug/mlruns/0/004574d1587d4d54b0c89ee95d32b0df/artifacts

### Code to reproduce issue

```python
mlflow.autolog(log_models=False)
func(*args, **kwargs)
mlflow.autolog(disable=True)
```

### Stack trace

No error observed by now

### What component(s) does this bug affect?

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/pipelines`: Pipelines, Pipeline APIs, Pipeline configs, Pipeline Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [X] `area/tracking`: Tracking Service, tracking client APIs, autologging

### What interface(s) does this bug affect?

- [X] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

### What language(s) does this bug affect?

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

### What integration(s) does this bug affect?

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

**Comment by harupy** [comment]
@PytLab Can you update the code reproduce the issue? It's not runnable.

**Comment by PytLab** [comment]
Sorry for that. I integrated mlflow code into the functions of my lib, let me extract the logic. 

I just want to enable the autolog of mlflow before the regressor fitting. I'm the autolog in this way:

```python
from xgboost import XGBRegressor
regressor = XGBRegressor()

# setting mlflow tracking uri code

mlflow.autolog(log_models=False)
regressor.fit(X, y)
mlflow.autolog(disable=True)
```
but from the tracking UI, there is no `train_mse` metric tracked. Is this expected?

**Comment by PytLab** [comment]
I noticed that 
> The [scikit-learn API](https://xgboost.readthedocs.io/en/latest/python/python_api.html#module-xgboost.sklearn) is not supported.

is this related?

**Comment by PytLab** [comment]
Will try with non-sklearn xgboost api with mlflow. Close as not a bug
